src/qr.rs

```diff
@@ -343,34 +343,55 @@ fn decode_webrtc_instance(_context: &Context, qr: &str) -> Result<Qr> {
 }
 
 #[derive(Debug, Deserialize)]
-struct CreateAccountResponse {
+struct CreateAccountSuccessResponse {
     email: String,
     password: String,
 }
+#[derive(Debug, Deserialize)]
+struct CreateAccountErrorResponse {
+    reason: String,
+}
 
 /// take a qr of the type DC_QR_ACCOUNT, parse it's parameters,
 /// download additional information from the contained url and set the parameters.
 /// on success, a configure::configure() should be able to log in to the account
 #[allow(clippy::indexing_slicing)]
 async fn set_account_from_qr(context: &Context, qr: &str) -> Result<()> {
     let url_str = &qr[DCACCOUNT_SCHEME.len()..];
+    let response = reqwest::Client::new().post(url_str).send().await?;
+
+    if response.status().is_success() {
+        let CreateAccountSuccessResponse { password, email } =
+            response.json().await.with_context(|| {
+                format!(
+                    "Cannot create account, response from {:?} is malformed",
+                    url_str
+                )
+            })?;
+        context.set_config(Config::Addr, Some(&email)).await?;
+        context.set_config(Config::MailPw, Some(&password)).await?;
 
-    let parsed: CreateAccountResponse = reqwest::Client::new()
-        .post(url_str)
-        .send()
-        .await?
-        .json()
-        .await
-        .with_context(|| format!("Cannot create account, request to {:?} failed", url_str))?;
-
-    context
-        .set_config(Config::Addr, Some(&parsed.email))
-        .await?;
-    context
-        .set_config(Config::MailPw, Some(&parsed.password))
-        .await?;
-
-    Ok(())
+        Ok(())
+    } else {
+        let response_text = response
+            .text()
+            .await
+            .with_context(|| format!("Cannot create account, request to {:?} failed", url_str))?;
+
+        match serde_json::from_str::<CreateAccountErrorResponse>(&response_text) {
+            Ok(error) => Err(anyhow!(error.reason)),
+            Err(parse_error) => {
+                context.emit_event(EventType::Error(format!(
+                    "Cannot create account, server response could not be parsed:\n{:#}\nraw response:\n{}",
+                    parse_error, response_text
+                )));
+                bail!(
+                    "Cannot create account, unexpected server response:\n {:?}",
```

Comment:
```suggestion
                    "Cannot create account, unexpected server response:\n{:?}",
```

(unless the space at the beginning of the line is intended)